lib/main_publiccloud.pm

```diff
@@ -77,47 +72,53 @@ sub load_publiccloud_consoletests {
     loadtest 'console/libgcrypt';
 }
 
-sub load_latest_publiccloud_tests {
+sub load_shared_publiccloud_tests {
     my $args = OpenQA::Test::RunArgs->new();
-    if (get_var('PUBLIC_CLOUD_PREPARE_TOOLS')) {
-        loadtest "publiccloud/prepare_tools";
-    }
-    elsif (get_var('PUBLIC_CLOUD_IMG_PROOF_TESTS')) {
-        loadtest "publiccloud/img_proof", run_args => $args;
-    }
-    elsif (get_var('PUBLIC_CLOUD_LTP')) {
-        loadtest 'publiccloud/run_ltp', run_args => $args;
-    }
-    elsif (get_var('PUBLIC_CLOUD_SLES4SAP')) {
-        loadtest 'publiccloud/sles4sap';
-    }
-    elsif (get_var('PUBLIC_CLOUD_ACCNET')) {
-        loadtest 'publiccloud/az_accelerated_net';
-    }
-    elsif (get_var('PUBLIC_CLOUD_CHECK_BOOT_TIME')) {
-        loadtest "publiccloud/boottime", run_args => $args;
-    }
-    elsif (get_var('PUBLIC_CLOUD_FIO')) {
-        loadtest 'publiccloud/storage_perf', run_args => $args;
-    }
-    elsif (get_var('PUBLIC_CLOUD_CONSOLE_TESTS')) {
-        loadtest "publiccloud/prepare_instance", run_args => $args;
-        loadtest "publiccloud/register_system", run_args => $args;
-        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-        load_extra_tests_prepare();
-        load_publiccloud_consoletests();
-        loadtest("publiccloud/ssh_interactive_end", run_args => $args);
-    }
-    elsif (get_var('PUBLIC_CLOUD_CONTAINERS')) {
-        loadtest "publiccloud/prepare_instance", run_args => $args;
-        loadtest "publiccloud/register_system", run_args => $args;
-        loadtest "publiccloud/ssh_interactive_start", run_args => $args;
-        load_podman_tests();
-        load_docker_tests();
-        loadtest("publiccloud/ssh_interactive_end", run_args => $args);
+    return -1 if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS') || get_var('PUBLIC_CLOUD_CONTAINERS'));
```

Comment:
why not just `0`? Normally, perl returns 1 if function is executed without errors. Just use `0`, and then you can do
`unless (load_shared_publiccloud_tests())`

Reply:
The reason is that i needed another value to distinguish  *PUBLIC_CLOUD_CONSOLE_TESTS* / *PUBLIC_CLOUD_CONTAINERS* and the rest (at least in the current sequence) . 

`unless (load_shared_publiccloud_tests())` behaves as such (with `return 0 if (get_var('PUBLIC_CLOUD_CONSOLE_TESTS') || get_var('PUBLIC_CLOUD_CONTAINERS'));`):
- consoletests - condition returns 0 and has correct scheduling
-  ltp                - calls ` loadtest('publiccloud/run_ltp)` and returns likely non-zero value and the scheduling adds ` publiccloud/ssh_interactive_start`, `publiccloud/instance_overview` and `publiccloud/ssh_interactive_end`. 

if you want to know why it is so, take a look on the [documentation](https://perldoc.perl.org/perlsub) where it writes `If you specify no return value, the subroutine returns an empty list in list context, the undefined value in scalar context, or nothing in void context.`

this is one case. ofc i tried other things too. i found the solution i present quick simple. in any case i would have to answer  questions like why i return 1 or zero, if or unless, and etc. i use `-1` as a *no-success* value which it doesnt conflict with the `true`/`false` definitions in perl, which is used to instruct a specific action when it matches.